### DISCUSSION TITLE
Worker node health endpoint?

### DISCUSSION THREAD

**Discussion by nwmcsween** [siderolabs/talos#9568] [Q&A]
Wondering if there is a http query I could use to check for worker node health similar to the kube api endpoint? What I want to accomplish is checking if the worker node is up and running in terraform without the overall health of the cluster itself.

**Comment by smira** [comment]
There's no HTTP query, as Talos requires mTLS anyways, but using Talos API you can fetch `MachineStatus` resource from a node and see if it's ready or not.
